ScriptEditor/ScriptEditor.py

```diff
@@ -180,9 +181,107 @@ def setup(self):
+        self._maxReadinessAttempts = 30  # 30 attempts with 100ms intervals = 3 seconds max
+        self.editorView.evalJS(readinessScript)
+    
+    def _checkEditorReadiness(self):
+        """Internal method to handle readiness checking"""
+        self._readinessCheckAttempts += 1
+        if self._readinessCheckAttempts > self._maxReadinessAttempts:
+            print("Warning: Monaco editor readiness timeout, proceeding anyway")
+            self.setupEditorFeatures()
+            return
+        
+        # Try again after 100ms
+        def retryReadinessCheck():
+            readinessScript = """
+                (function() {
+                    try {
+                        if (window.monaco && window.editor && typeof window.editor.getModel === 'function' && window.editor.getModel()) {
+                            return 'ready';
+                        }
+                    } catch (e) {
+                        console.log('Editor readiness check error:', e);
+                    }
+                    return 'not-ready';
+                })();
+            """
+            self.editorView.evalJS(readinessScript)
+        qt.QTimer.singleShot(100, retryReadinessCheck)
+    
+    def _fallbackSetup(self):
+        """Fallback setup in case readiness detection fails"""
+        if not hasattr(self, '_editorSetupComplete') or not self._editorSetupComplete:
+            print("Editor readiness detection timed out, attempting fallback setup...")
+            self.setupEditorFeatures()
+    
+    def setupDragDropHandling(self):
+        """Setup drag and drop handling to prevent files from being dropped directly into Monaco"""
+        # First, try to disable drag and drop at the QWebView level
+        try:
+            # Get the underlying QWebView and disable its drag and drop
+            webView = self.editorView.webView()
+            if webView:
+                webView.setAcceptDrops(False)
+                print("Disabled drag and drop at QWebView level")
+        except Exception as e:
+            print(f"Could not disable QWebView drag and drop: {e}")
+        
+        # Instead of overriding methods, use an event filter
+        try:
+            self.dragDropEventFilter = DragDropEventFilter(self)
+            self.editorView.installEventFilter(self.dragDropEventFilter)
+            print("Installed drag and drop event filter")
+        except Exception as e:
+            print(f"Could not install event filter: {e}")
+        
+        # Also try to disable accepts drops entirely
+        try:
+            self.editorView.setAcceptDrops(False)
+            print("Disabled acceptDrops on editor view")
+        except Exception as e:
+            print(f"Could not disable acceptDrops: {e}")
+        
+        # Simple drag and drop blocking for Monaco (less aggressive)
+        dragDropScript = """
+        console.log('Setting up simple drag and drop blocking...');
+        """
```

Comment:
It would be great to format this code for readability even if it's not needed syntactically.